hw/ip/spi_device/dv/base_sim_cfg.hjson

```diff
@@ -47,6 +47,15 @@
   // Enable cdc instrumentation.
   run_opts: ["+cdc_instrumentation_enabled=1"]
 
+  build_modes : {
+    spi_device_1r1w: {
+      build_opts: ["+define+SRAM_TYPE=spi_device_pkg::SramType1r1w"]
+    }
+    spi_device_2p: {
+      build_opts: ["+define+SRAM_TYPE=spi_device_pkg::SramType2p"]
+    }
+  }
```

Comment:
Can you remove the build modes and pipe the plusdefine in via buildopts as done for lc_ctrl?

https://github.com/lowRISC/opentitan/blob/7e196e6f072791aa17817b7691aed6b258399eeb/hw/ip/lc_ctrl/dv/lc_ctrl_volatile_unlock_disabled_sim_cfg.hjson#L13-L14

I ran into issues collecting coverage with this variant previously.

Reply:
Yep, happily. I've just pushed up a version that I *think* should work. Leaving this for you to glance at before it gets merged but please feel free to hit merge if it looks sensible to you.